## Exercicios Módulo Python - Universidade dos Dados.

### Nesse dataset, temos as informações de diversos pacientes (gênero, se é fumante, imc, etc) e seus gastos com saúde.

1) Importe os dados e verifique as primeiras linhas do dataset. Verifique também as últimas linhas do dataset.

2) Verifique se existe algum valor missing no dataset.

3) Quanto de memória o dataset está consumindo?

4) Gere as estatísticas descritivas das colunas numéricas do dataset. Qual a mediana da coluna BMI? E a média? Qual desses valores é maior? O que isso significa? Caso não saiba o que isso significa, passe os 2 valores ao chatGPT e pergunte a ele o que significa a média ser maior ou menor que a mediana.

5) Qual a proporção de fumantes no nosso dataset? E de não fumantes?

6) Existe alguma diferença na cobrança para fumantes e não fumantes? Se sim, ao que você imagina que isso se deve?

7) Qual a média de BMI para fumantes? Ela é muito diferente da observada para fumantes? Se sim, por qual motivo você acha que isso ocorre?

8) Existe alguma diferença de cobrança para quem tem filhos? Identifique se ter filhos parece acarretar numa cobrança maior olhando a média de charges para quem tem 0 filhos e a média para quem tem 1 ou mais filhos.

9) Para nós seres humanos, o BMI de 20 a 24,9 é considerado normal. Acima disso, é considerado acima do peso, abaixo é considerado abaixo do peso. Crie uma nova coluna, chamada "bmi_male" que receberá o texto "abaixo do normal" para BMI abaixo de 20, "normal" para BMI de 20 a 24,9 e "acima do normal" para BMI acima de 24,9.

10) Se uma pessoa é fumante e ainda tem um BMI acima do considerado normal, é possível que isso seja um indicador negativo para a seguradora. Quantos homens estão nessa situação?

In [1]:
import pandas as pd

In [10]:
# Importando os dados
df = pd.read_csv("C:\Portfolio\Projeto-Estudo\insurance.csv")

In [15]:
# Exercício 1 - Retorne as primeiras e as últimas linhas do dataset.
df.head() 

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [16]:
df.tail()

age     sex    bmi  children smoker     region     charges
1333   50    male  30.97         3     no  northwest  10600.5483
1334   18  female  31.92         0     no  northeast   2205.9808
1335   18  female  36.85         0     no  southeast   1629.8335
1336   21  female  25.80         0     no  southwest   2007.9450
1337   61  female  29.07         0    yes  northwest  29141.3603

In [18]:
# Exercício 2 - Verifique se existe algum valor missing no dataset
# Exercício 3 - Quanto de memória o dataset está consumindo?
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


Não existe nenhum valor missing no dataset e o mesmo ocupa 73.3KB de memória

In [19]:
# Exercício 4 - Gere as estatísticas descritivas das colunas numéricas do dataset. 
# Qual a mediana da coluna BMI? E a média? Qual desses valores é maior? O que isso significa? 
df.describe() 

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

- A mediana da coluna BMI (ou IMC) é de 30.40
- A média da coluna BMI (ou IMC) é de 30.66
- Comparando os valores da mediana com a média, notamos que o valor da média é maior, o que significa que há valores altos de BMI que estão influenciando a média.

In [20]:
# Exercício 5 - Qual a proporção de fumantes no nosso dataset? E de não fumantes?
df['smoker'].value_counts()

no     1064
yes     274
Name: smoker, dtype: int64

In [21]:
df['smoker'].value_counts(normalize=True)

no     0.795217
yes    0.204783
Name: smoker, dtype: float64

No dataset temos 1.604 não fumantes (79,52%) e 274 fumantes (20,48%).

In [33]:
# Exercício 6 - Existe alguma diferença na cobrança para fumantes e não fumantes? Se sim, ao que você imagina que isso se deve?
df.groupby("smoker")[["charges"]].mean()

charges
smoker              
no       8434.268298
yes     32050.231832

Sim, existe uma diferença significativa na média da cobrança para fumantes e não fumantes, possivelmente pelo risco que o tabagismo oferece a saúde.

In [36]:
# Exercício 7 - Qual a média de BMI para fumantes? 
# Ela é muito diferente da observada para fumantes? Se sim, por qual motivo você acha que isso ocorre?
df.groupby("smoker")[["bmi"]].mean()

bmi
smoker           
no      30.651795
yes     30.708449

- A média de BMI para fumantes é de 30.65
- Há pouca diferença entre a média de BMI para fumantes e não fumantes. Isso pode significar que por mais que o tabagismo seja prejudicial à saúde, ele não internefere muito no BMI das pessoas.

In [78]:
# Exercício 8 - Existe alguma diferença de cobrança para quem tem filhos? 
# Identifique se ter filhos parece acarretar numa cobrança maior olhando a média de charges para quem tem 0 filhos 
# e a média para quem tem 1 ou mais filhos
media_cobranca_sem_filhos = df[df['children'] == 0]['charges'].mean()
media_cobranca_com_filhos = df[df['children'] >= 1]['charges'].mean()
diferenca_media_cobranca = media_cobranca_com_filhos - media_cobranca_sem_filhos

print(f"Média de cobrança para quem não tem filhos: {media_cobranca_sem_filhos:.2f}")
print(f"Média de cobrança para quem tem 1 ou mais filhos: {media_cobranca_com_filhos:.2f}")
print(f"Diferença da média do valor de cobrança: {diferenca_media_cobranca:.2f}")

Média de cobrança para quem não tem filhos: 12365.98
Média de cobrança para quem tem 1 ou mais filhos: 13949.94
Diferença da média do valor de cobrança: 1583.97


Olhando o valor das médias de cobrança e o valor da diferença das médias, chegamos a conclusão de que ter filhos acarreta em um valor de cobrança mais alto.

In [86]:
# Exercício 9 - Para o homem, o BMI de 20 a 24,9 é considerado normal. 
# Acima disso, é considerado acima do peso, abaixo é considerado abaixo do peso. 
# Crie uma nova coluna, chamada "bmi_male" que receberá o texto "abaixo do normal" para BMI abaixo de 20, 
# "normal" para BMI de 20 a 24,9 e "acima do normal" para BMI acima de 24,9.
import numpy as np

df['bmi_male'] = np.where(
    df['bmi'] < 20, "abaixo do normal",
    np.where(
        (df['bmi'] >= 20) & (df['bmi'] <= 24.9), "normal",
        "acima do normal"
    )
)

df.head(30)

age     sex     bmi  children smoker     region      charges  \
0    19  female  27.900         0    yes  southwest  16884.92400   
1    18    male  33.770         1     no  southeast   1725.55230   
2    28    male  33.000         3     no  southeast   4449.46200   
3    33    male  22.705         0     no  northwest  21984.47061   
4    32    male  28.880         0     no  northwest   3866.85520   
5    31  female  25.740         0     no  southeast   3756.62160   
6    46  female  33.440         1     no  southeast   8240.58960   
7    37  female  27.740         3     no  northwest   7281.50560   
8    37    male  29.830         2     no  northeast   6406.41070   
9    60  female  25.840         0     no  northwest  28923.13692   
10   25    male  26.220         0     no  northeast   2721.32080   
11   62  female  26.290         0    yes  southeast  27808.72510   
12   23    male  34.400         0     no  southwest   1826.84300   
13   56  female  39.820         0     no  southeast  11090.71780   
14   27    male  42.130         0    yes  southeast  39611.75770   
15   19    male  24.600         1     no  southwest   1837.23700   
16   52  female  30.780         1     no  northeast  10797.33620   
17   23    male  23.845         0     no  northeast   2395.17155   
18   56    male  40.300         0     no  southwest  10602.38500   
19   30    male  35.300         0    yes  southwest  36837.46700   
20   60  female  36.005         0     no  northeast  13228.84695   
21   30  female  32.400         1     no  southwest   4149.73600   
22   18    male  34.100         0     no  southeast   1137.01100   
23   34  female  31.920         1    yes  northeast  37701.87680   
24   37    male  28.025         2     no  northwest   6203.90175   
25   59  female  27.720         3     no  southeast  14001.13380   
26   63  female  23.085         0     no  northeast  14451.83515   
27   55  female  32.775         2     no  northwest  12268.63225   
28   23    male  17.385         1     no  northwest   2775.19215   
29   31    male  36.300         2    yes  southwest  38711.00000   

            bmi_male  
0    acima do normal  
1    acima do normal  
2    acima do normal  
3             normal  
4    acima do normal  
5    acima do normal  
6    acima do normal  
7    acima do normal  
8    acima do normal  
9    acima do normal  
10   acima do normal  
11   acima do normal  
12   acima do normal  
13   acima do normal  
14   acima do normal  
15            normal  
16   acima do normal  
17            normal  
18   acima do normal  
19   acima do normal  
20   acima do normal  
21   acima do normal  
22   acima do normal  
23   acima do normal  
24   acima do normal  
25   acima do normal  
26            normal  
27   acima do normal  
28  abaixo do normal  
29   acima do normal

In [95]:
# Exercício 10 - Se uma pessoa é fumante e ainda tem um BMI acima do considerado normal, é possível que isso seja um indicador 
# negativo para a seguradora. Quantas pessas estão nessa situação?
pessoas_fumantes_bmi_alto = df[(df['smoker'] == 'yes') & (df['bmi'] > 24.9)]
contagem_pessoas_indicador_negativo = len(pessoas_fumantes_bmi_alto)

print(f"{contagem_pessoas_indicador_negativo} pessoas estão na situação de um possível indicador negativo para a seguradora.")

219 pessoas estão na situação de um possível indicador negativo para a seguradora.
